### ISSUE TITLE
False positive: snake_case warning for operators

### ISSUE THREAD

**Issue by hissssst** [rrrene/credo#1026]
### Environment

```sh
$ mix credo -v
1.6.7-ref.main.2245c20+uncommittedchanges

$ elixir -v
Erlang/OTP 24 [erts-12.3.2.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.14.2 (compiled with Erlang/OTP 24)

$ uname -som
Linux x86_64 GNU/Linux
```

### What were you trying to do?

```elixir
defmodule X do
  import Kernel, except: [&&: 2]

  def x && x, do: true
  def _ && _, do: false
end
```

This is valid code, since is redefines the `&&` operator

### Expected outcome

No warnings, since the code above just redefines the `&&` operator, which is a pair of special symbols, thus snake case check makes no sense for it.

### Actual outcome

```
┃ [R] ↗ Function/macro/guard names should be written in snake_case.
┃       lib/tria/language/ternary.ex:30 #(Tria.Language.Ternary.||)
┃ [R] ↗ Function/macro/guard names should be written in snake_case.
┃       lib/tria/language/ternary.ex:26 #(Tria.Language.Ternary.&&)
```

**Comment by rrrene** [comment]
I could not reproduce this (and I added a test to try to reproduce this in CI).

**Comment by hissssst** [comment]
Oh, yes. This happens only when operator is defined with guard. I've reproduced the bug in this example
```elixir
defmodule Xxx do

  defguard is_ternary(x) when x in ~w[yes no maybe]a

  def d && d when is_ternary(x), do: d

end
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by marcandre** [comment]
Had same issue, works with `master`.

A release would be appreciated.

**Comment by rrrene** [comment]
This is released in `1.7.0-rc.2`, will release `1.7.0` shortly.

**Comment by hissssst** [comment]
Fixed in master
